### PR TITLE
[DALA-957] Swagger Documents

### PR DESCRIPTION
**Pull request by nomad3k**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/service-search-api.yaml [hunk 1]
```diff
@@ -18,4 +157,483 @@ servers:
...
+        - name: api-version
+          description: The API version number can be supplied as a query parameter.
+          in: query
+          schema:
+            type: number
+            default: 1
+            enum: [1]
+        - name: search
+          in: query
+          description: The text to search for. All searchable fields are searched by default unless searchFields is specified. When searching searchable fields, the search text itself is tokenized, so multiple terms can be separated by white space (for example search=hello world). To match any term, use * (this can be useful for boolean filter queries). Omitting this parameter has the same effect as setting it to *.
```
Comment (michaelthemoon) [inline specification/service-search-api.yaml:181]: Is this actually a default * then?

File: specification/service-search-api.yaml [hunk 2]
```diff
@@ -18,4 +157,483 @@ servers:
...
+        - name: queryType
+          description: Defaults to simple. When set to simple, search text is interpreted using a simple query language that allows for symbols such as +, * and "". When the query type is set to full, search text is interpreted using the Lucene query language which allows field-specific and weighted searches.
+          in: query
+          schema:
+            type: string
+            enum:
+            - 'simple'
+            - 'full'
+            default: 'simple'
+        - name: $skip
```
Comment (michaelthemoon) [inline specification/service-search-api.yaml:207]: Is this default 0?

File: specification/service-search-api.yaml [hunk 3]
```diff
@@ -18,4 +157,483 @@ servers:
...
+
+    CoordinateReferenceSystem:
+      description: Co-ordinate reference system
+      type: object
+      properties:
+        "type":
+          type: string
+        "properties":
+          type: object
+          additionalProperties: true
```
Comment (michaelthemoon) [inline specification/service-search-api.yaml:638]: Do we know what these properties can be? If not we should find out what they are.

File: specification/examples/organisation_v1.json [hunk 4]
```diff
@@ -0,0 +1,6923 @@
+{
```
Comment (michaelthemoon) [inline specification/examples/organisation_v1.json:1]: Is this copy pasted from somewhere. Glazing over looking at this one.

**Review by michaelthemoon — APPROVED** [review]

File: specification/service-search-api.yaml (hunk 1 above)
Comment (nomad3k) [reply specification/service-search-api.yaml:181]: No, this is a constant.  To access this version of the api you need to do `api-version=1`.  Swagger doesn't support JSON schema `const` and this is the work around. 

```
enum: [1]
default: 1
```

File: specification/service-search-api.yaml [hunk 5]
```diff
@@ -18,4 +157,483 @@ servers:
...
+            - 'any'
+            - 'all'
+            default: 'any'
+        - name: searchFields
+          in: query
+          description: The list of comma-separated field names to search for the specified text.
+          schema:
+            type: string
+        - name: queryType
+          description: Defaults to simple. When set to simple, search text is interpreted using a simple query language that allows for symbols such as +, * and "". When the query type is set to full, search text is interpreted using the Lucene query language which allows field-specific and weighted searches.
```
[the remark this answers is not in the document]
Comment (nomad3k) [reply specification/service-search-api.yaml:199]: This is lifted from the original documentation.

File: specification/service-search-api.yaml (hunk 3 above)
Comment (nomad3k) [reply specification/service-search-api.yaml:639]: We do.  It's a massively complex Azure Search query, which is documented in the description.

File: specification/examples/organisation_v1.json (hunk 4 above)
Comment (nomad3k) [reply specification/examples/organisation_v1.json:1]: This is live output from the API.

File: specification/service-search-api_v2.yaml [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+    | Legacy      | https://api.nhs.uk/service-search/              |
+    | New         | https://api.service.nhs.uk/service-search-api/  |
+    | Sandbox     | https://sandbox.api.service.nhs.uk/service-search-api/  |
+
+
+    ### Sandbox testing
+
+    Our sandbox environment:
+    
+    is for early developer testing
```
[the remark this answers is not in the document]
Comment (nomad3k) [reply specification/service-search-api_v2.yaml:93]: Hmm.. it also doesn't exist. :)

[the remark this answers is not in the document]
Comment (nomad3k) [reply specification/service-search-api_v2.yaml:93]: I need to create the sandbox, and will update the documentation when it's available.

File: specification/service-search-api.yaml (hunk 2 above)
Comment (nomad3k) [reply specification/service-search-api.yaml:192]: 0?  I don't follow.

Comment (michaelthemoon) [reply specification/service-search-api.yaml:192]: The comment is for the block contained in $skip (line 207)

File: specification/service-search-api.yaml (hunk 1 above)
Comment (michaelthemoon) [reply specification/service-search-api.yaml:166]: Line on 181. Description basically says it is a default value but we have no default field

Comment (nomad3k) [reply specification/service-search-api.yaml:166]: Yeah, I've done that now.  Not needed but clarifies it.

**Review by michaelthemoon — APPROVED** [review]

**Pull request merged**